### PR TITLE
Clarify JSONPath expression syntax

### PR DESCRIPTION
**Pull request by glyn**
Fixes https://github.com/ietf-wg-jsonpath/draft-ietf-jsonpath-base/issues/377

### CODE REVIEW

File: draft-ietf-jsonpath-base.md [hunk 1]
```diff
@@ -622,7 +621,7 @@ of input node.
 ### Syntax
 {: unnumbered}
 
-Every JSONPath query MUST begin with the root identifier `$`.
+Every JSONPath query (except those inside filter expressions, see {{filter-selector}}) MUST begin with the root identifier `$`.
```
Comment (cabo) [inline draft-ietf-jsonpath-base.md:624]: Do we call the things inside the filter expressions (filter-path/singular-path) "JSONPath query"?

Comment (glyn) [reply draft-ietf-jsonpath-base.md:624]: Yes, I think we do. Section 1.4 says:
> A JSONPath expression consists of an identifier followed by a series of zero or more segments each of which contains one or more selectors.

and "query" is just a short name for "JSONPath expression".

Comment (cabo) [reply draft-ietf-jsonpath-base.md:624]: Hmm, wouldn't it be useful to have a name for the thing that can be a JSONPath query outside a filter expression?

Comment (goessner) [reply draft-ietf-jsonpath-base.md:624]: Hmm ... there is only a single JSONPath expression outside of filter expressions.

We also should obey our own terminology:

Query:   Short name for a JSONPath expression.

**Review by cabo — APPROVED** [review]

**Pull request merged**